bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/DLinkSmartHomeDiscoveryParticipant.java

```diff
@@ -96,7 +97,11 @@ private ThingTypeUID getThingType(final ServiceInfo serviceInfo) {
 
     private DiscoveryResult createMotionSensor(final ThingUID thingUID, final ThingTypeUID thingType,
             final ServiceInfo serviceInfo) {
-        final String host = serviceInfo.getHostAddresses()[0];
+        final Inet4Address[] addresses = serviceInfo.getInet4Addresses();
+        if (addresses.length != 1) {
+            return null;
+        }
+        final String host = addresses[0].getHostAddress();
```

Comment:
This is a change in the discovery that you even don't announce ?
I think this should be mentioned in the description of your PR.

Reply:
Description updated to include why this change was made.